### PR TITLE
mark tests that are meant for goto-gcc only

### PR DESCRIPTION
**Pull request by kroening**

### CODE REVIEW

**Review by tautschnig — APPROVED** [review]

**Review by smowton — APPROVED** [review]

File: regression/goto-gcc/gcc_attributes10/main.c [hunk 1]
```diff
@@ -4,13 +4,13 @@
 #ifdef __GNUC__
 #ifndef __clang__
 
-int  x   __attribute__ ((__vector_size__ (12), __may_alias__));
+int  x   __attribute__ ((__vector_size__ (16), __may_alias__));
```
Comment (tautschnig) [inline regression/goto-gcc/gcc_attributes10/main.c]: What's the point of this change?

Comment (kroening) [reply regression/goto-gcc/gcc_attributes10/main.c]: To make this pass though gcc; gcc wants powers of two for the vector size.

Comment (tautschnig) [reply regression/goto-gcc/gcc_attributes10/main.c]: Uh, ok, could this please be recorded in the commit message?

File: regression/goto-gcc/gcc_builtins4/test.desc [hunk 2]
```diff
@@ -1,4 +1,4 @@
-CORE
+KNOWNBUG
```
Comment (tautschnig) [inline regression/goto-gcc/gcc_builtins4/test.desc]: The commit message says "fix __builtin_classify_type", which is quite confusing in light of this:
1) Please add an explanation of what is being fixed/what the problem was before;
2) moving from `CORE` to `KNOWNBUG` seems contradictory with "fix ..."

**Review by tautschnig — CHANGES_REQUESTED** [review]
Requesting clarification.

File: regression/goto-gcc/gcc_attributes10/main.c (hunk 1 above)
Comment (kroening) [reply regression/goto-gcc/gcc_attributes10/main.c]: done

File: regression/goto-gcc/gcc_builtins4/main.c [hunk 3]
```diff
@@ -7,6 +7,7 @@ union { int i; } u;
 enum { Econst } e;
 int a[10];
 
+#ifndef __clang__
```
Comment (tautschnig) [inline regression/goto-gcc/gcc_builtins4/main.c]: I'd appreciate if all the `#ifndef __clang__` changes could be placed in a separate commit; at the bare minimum there should be some information in the commit message.

File: regression/Makefile [hunk 4]
```diff
@@ -14,6 +14,7 @@ DIRS = cbmc \
        goto-diff \
        test-script \
        goto-analyzer-taint \
+       goto-cc \
```
Comment (tautschnig) [inline regression/Makefile]: As discussed out-of-band: please explain this in the commit message and/or rename the directory to clarify that these are GCC/Clang-only tests as they use GCC/Clang-only command-line options.

File: src/ansi-c/c_typecheck_expr.cpp [hunk 5]
```diff
@@ -2594,22 +2595,34 @@ exprt c_typecheck_baset::do_special_functions(
...
+
+    unsigned type_number;
+
+    if(type.id()==ID_bool || type.id()==ID_c_bool)
+    {
+      // clang returns 4 for _Bool, gcc treats these as 'int'.
+      if(config.ansi_c.preprocessor == configt::ansi_ct::preprocessort::CLANG)
+        type_number=4;
+      else
+        type_number=1;
```
Comment (tautschnig) [inline src/ansi-c/c_typecheck_expr.cpp]: Nit picking: I think I'd use a ternery `?:` here.

File: src/ansi-c/c_typecheck_expr.cpp [hunk 6]
```diff
@@ -2594,22 +2595,34 @@ exprt c_typecheck_baset::do_special_functions(
...
-    // gcc treats these as 'int'
+    typet type=follow(object.type());
+
+    // use underlying type for bit fields
+    if(type.id()==ID_c_bit_field)
+      type=to_c_bit_field_type(type).subtype();
+
+    unsigned type_number;
+
+    if(type.id()==ID_bool || type.id()==ID_c_bool)
```
Comment (tautschnig) [inline src/ansi-c/c_typecheck_expr.cpp]: Nit picking: clang-format style spacing around `==` and `=` (across this commit).

**Review by tautschnig — APPROVED** [review]

File: src/ansi-c/c_typecheck_expr.cpp (hunk 5 above)
Comment (kroening) [reply src/ansi-c/c_typecheck_expr.cpp]: done

File: src/ansi-c/c_typecheck_expr.cpp (hunk 6 above)
Comment (kroening) [reply src/ansi-c/c_typecheck_expr.cpp]: done

**Review by peterschrammel — APPROVED** [review]
